### PR TITLE
Use `dag_maker` fixture in models/test_taskinstance.py

### PR DESCRIPTION
**Pull request by ephraimbuddy**
The change applies dag_maker fixture in test_taskinstance.py

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: tests/models/test_taskinstance.py [hunk 1]
```diff
@@ -437,75 +442,51 @@ def create_task_instance():
...
-        with models.DAG(dag_id='test_run_pooling_task') as dag:
-            task2 = DummyOperator(
-                task_id='test_run_pooling_task_op',
-                owner='airflow',
-                executor_config={'bar': 'baz'},
-                start_date=timezone.datetime(2016, 2, 1, 0, 0, 0),
-            )
+        task2 = DummyOperator(
+            task_id='test_run_pooling_task_op2',
+            owner='airflow',
```
Comment (ashb) [inline tests/models/test_taskinstance.py]:
```suggestion
```

Let's remove these -- it's not relevant to the any of these test I think.

Comment (ashb) [reply tests/models/test_taskinstance.py]: Looks like you already did get rid of all the others :+1:

**Review by ashb — APPROVED** [review]
LGTM.

File: tests/models/test_taskinstance.py [hunk 2]
```diff
@@ -1969,9 +1836,9 @@ def test_get_rendered_template_fields(self):
 
     @mock.patch.dict(os.environ, {"AIRFLOW_IS_K8S_EXECUTOR_POD": "True"})
     @mock.patch("airflow.settings.pod_mutation_hook")
-    def test_render_k8s_pod_yaml(self, pod_mutation_hook):
-        with DAG('test_get_rendered_k8s_spec', start_date=DEFAULT_DATE):
-            task = BashOperator(task_id='op1', bash_command="{{ task.task_id }}")
+    def test_render_k8s_pod_yaml(self, pod_mutation_hook, dag_maker):
+        with dag_maker('test_get_rendered_k8s_spec'):
+            task = BashOperator(task_id='op1', bash_command="{{ taTsk.task_id }}")
```
Comment (ephraimbuddy) [inline tests/models/test_taskinstance.py]: The tests are failing here, any ideas?

File: tests/models/test_taskinstance.py [hunk 3]
```diff
@@ -2021,9 +1879,9 @@ def test_render_k8s_pod_yaml(self, pod_mutation_hook):
 
     @mock.patch.dict(os.environ, {"AIRFLOW_IS_K8S_EXECUTOR_POD": "True"})
     @mock.patch.object(RenderedTaskInstanceFields, 'get_k8s_pod_yaml')
-    def test_get_rendered_k8s_spec(self, rtif_get_k8s_pod_yaml):
+    def test_get_rendered_k8s_spec(self, rtif_get_k8s_pod_yaml, dag_maker):
```
Comment (ephraimbuddy) [inline tests/models/test_taskinstance.py:1918]: This test is failing, any ideas?

**Review by ashb — APPROVED** [review]

File: tests/conftest.py [hunk 4]
```diff
@@ -480,6 +480,7 @@ def __call__(self, dag_id='test_dag', session=None, **kwargs):
                     self.start_date = DEFAULT_DATE
             self.kwargs['start_date'] = self.start_date
             self.dag = DAG(dag_id, **self.kwargs)
+            self.dag.fileloc = request.module.__file__
```
Comment (ashb) [inline tests/conftest.py:483]: Oh nice and easy :D

Comment (ephraimbuddy) [reply tests/conftest.py:483]: Thanks for the idea!

**Review by ashb — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by ephraimbuddy** [comment]
This test is flaky in the CI  `TestSchedulerJob.test_add_unparseable_file_before_sched_start_creates_import_error`, any ideas @potiuk

**Pull request merged**